Return bytes sent on Publish function

Florisheinen1 opened the pull request:
@rolfvdhulst found a [small bug in RobotHub](https://github.com/RoboTeamTwente/roboteam_robothub/issues/66). A nice way to solve this would be to return the bytes sent instead of success when a publish function is called. This PR makes that change.
Other repos do depend on this mechanic, so this PR is co-dependent on:
- The [PR in AI](https://github.com/RoboTeamTwente/roboteam_ai/pull/1399)
- The [PR in RobotHub](https://github.com/RoboTeamTwente/roboteam_robothub/pull/67)
- The [PR in World](https://github.com/RoboTeamTwente/roboteam_world/pull/80)

Florisheinen1 committed:
Return bytes sent on Publish function

rolfvdhulst commented on src/RobotCommandsNetworker.cpp:
```diff
@@ -85,9 +85,10 @@ rtt::RobotCommands protoToRobotCommands(const proto::RobotCommands& protoCommand
     auto protoRobotCommands = robotCommandsToProto(commands);
-    return this->send(protoRobotCommands.SerializeAsString());
+
+    return this->send(protoRobotCommands.SerializeAsString()) ? protoRobotCommands.ByteSizeLong() : -1;
```
-1 of std::size_t will just underflow: maybe prefer using std::numeric_types<std::size_t>::max() in that case? That is a bit cleaner as -1 is a conversion.

Florisheinen1 replied:
I changed it to returning 0 on failure. This perhaps also makes more sense, as now you know that 0 bytes, a.k.a. nothing, has been sent. Now the underflow problem should not occur, and you can easily check with the `>` operator if your message has been sent

rolfvdhulst reviewed (commented):
I am not a big fan of using -1 with std::size_t (replace with std::numeric_limits<std::size_t>::max()? ) OR maybe do something else...

Florisheinen1 committed:
Improve efficiency of calculating bytes sent

Asking ByteSizeLong on a proto object effectively serializes it. This means every object is serialized twice. To prevent this, I just directly return the size of the serialized object

Florisheinen1 committed:
Return 0 bytes sent on failure

Returning -1 could result in overflow, resulting in the mechanic of checking if bytes sent < 0 to be always true. Returning 0 solves the overflow, so checking can be done if bytes sent > 0 for success

alexander-de-ranitz reviewed (approved):
LGTM
